releasenotes/notes/deprecate-aer-hooks-1bc00a96ebb269f6.yaml

```diff
@@ -0,0 +1,14 @@
+---
+deprecations:
+  - |
+    Use of the :attr:`qiskit.Aer` object is deprecated and will be removed in Qiskit 1.0.  You
+    should instead use the same object from the ``qiskit_aer`` namespace, which is a drop-in
+    replacement.
+  - |
+    Importing from ``qiskit.providers.aer`` is deprecated and will stop working in Qiskit 1.0.  You
```

Comment:
Did we have a deprecation warning for the namespace hooks in general? Not that it hurts to do this explicitly because we're deprecating the backwards compat import machinery in this PR.

Reply:
What kind of statements do you mean, sorry?

If you mean the objects in `qiskit.namespace` themselves, I haven't put deprecations on them a) we need to use them for the whole length of their existence so would trigger warnings by our regular use of them (or we'd have to wait another cycle to deprecate), and b) they're not public per our policy (there's no documentation of `qiskit.namespace`).

Reply:
For linking purposes: at the time, we discussed whether to make `qiskit.namespace` explicitly private, but equally didn't make a positive decision that it would be _public_ per the deprecation policy, we just didn't call it `qiskit._namespace`.  By [the intended current terms of the deprecation policy](https://github.com/Qiskit/documentation/pull/366/), it doesn't count as public because it's not documented, though caveat: we never merged #11205 on Terra, and then the deprecation policy got moved to `DEPRECATIONS.md` on Terra in #11218 in its old form, so we need to resurrect #11205 here.

Reply:
I meant https://github.com/Qiskit/qiskit/pull/11223 which implicitly covered `qiskit.providers.aer` at least in my head, the `qiskit.namespace` machinery was just an explicit exception to that.